src/main/java/com/nekomaster1000/infernalexp/client/DynamicLightingHandler.java

```diff
@@ -19,14 +20,14 @@
     public static void tick(LivingEntity entity) {
         if (entity != null && MinecraftInstance.player != null && MinecraftInstance.player.ticksExisted % (int) InfernalExpansionConfig.ClientConfig.LUMINOUS_REFRESH_RATE.getDouble() == 0) {
             if (shouldGlow(entity)) {
-                LIGHT_SOURCES.put(entity.getPosition(), new LightData());                
+                LIGHT_SOURCES.put(entity.getPosition(), new LightData(getTimeAmplifier(entity)));
             }
             if (entity == MinecraftInstance.player) {
                 LIGHT_SOURCES.forEach((pos, data) -> {
                     if (data.time == 0) {
                         data.shouldKeep = false;
                     }
-                    if (data.time == 20 || !data.shouldKeep) {
+                    if (data.time == (20 * data.amplifier) || !data.shouldKeep) {
```

Comment:
```suggestion
                    if (data.time == 20 * data.amplifier || !data.shouldKeep) {
```
These parentheses are useless, since `20` and `data.amplifier` will always be multiplied, and `20` will never be compared by itself.